src/helpers/screenings.js

```diff
@@ -3,13 +3,52 @@ import fetch from "node-fetch";
 const API = "https://plankton-app-xhkom.ondigitalocean.app/api/screenings";
 
 export async function movieScreenings(id) {
-    const res = await fetch(`${API}?filters[movie]=${id}`);
-    const payload = await res.json();
-    return payload.data;
+  const res = await fetch(`${API}?filters[movie]=${id}`);
+  const payload = await res.json();
+  return payload.data;
 }
 
 export async function getAllScreenings() {
-    const res = await fetch(API);
-    const payload = await res.json();
-    return payload.data;
+  const res = await fetch(API);
+  const payload = await res.json();
+  return payload.data;
+}
+
+export async function screeningsStartpage() {
+  const apiRes = await fetch(API + "?populate=movie");
+  const payload = await apiRes.json();
+  const result = payload.data
+    .map((item) => ({
+      id: item.id,
+      ...item.attributes,
+      movie: {
+        id: item.attributes.movie.data.id,
+        ...item.attributes.movie.data.attributes,
+      },
+    }))
+    .filter((screenings) => {
+      const getDatesBetweenDates = (startDate, endDate) => {
+        let dates = [];
+
+        const theDate = new Date(startDate);
+        while (theDate < new Date(endDate)) {
+          dates = [...dates, new Date(theDate).toISOString().split("T")[0]];
+          theDate.setDate(theDate.getDate() + 1);
+        }
+        dates = [...dates, new Date(endDate).toISOString().split("T")[0]];
+        return dates;
+      };
+      const start = new Date().toISOString().split("T")[0];
+      const end = new Date(new Date().getTime() + 4 * 24 * 60 * 60 * 1000)
+        .toISOString()
+        .split("T")[0];
+
+      for (let i = 0; i < getDatesBetweenDates(start, end).length; i++) {
```

Comment:
When you call a function like this inside the _condition_ of a for loop, it will be evaluated (and hence executed) every time the loop runs. So since it always returns five dates, it will be executed five times here.

It's also executed again on the next line, so that means it's executed 10 times.

That's for every screening, so if you retrieve 25 screenings, that's 10 * 25 times in total that `getDatesBetweenDates()` will be called.

Inside `getDatesBetweenDates()` there is another loop, that is 4 iterations long (if I'm not mistaken). That means that in total, the code on lines 35-36 is executed 1000 times (10 * 25 * 4) whenver `screeningsStartpage()` is called.

That's not a very efficient algorithm. But it works! 😊